mk/docker.mk

```diff
@@ -127,8 +127,12 @@ docker/purge: ## Dev: Remove all Docker containers, images, networks and volumes
 
 .PHONY: docker/login
 docker/login:
-	$(call GATE_PUSH,docker login -u $(DOCKER_USERNAME) -p $(DOCKER_API_KEY) $(DOCKER_REGISTRY))
+	# We need the host part of the registry
+	SERVER=$$(echo "$(DOCKER_REGISTRY)" | sed -e 's|\(.*\)/.*|\1|'); \
+	$(call GATE_PUSH,docker login -u $(DOCKER_USERNAME) -p $(DOCKER_API_KEY) $${SERVER})
```

Comment:
or maybe even just take out `docker.io` to separate variable:
```makefile
DOCKER_SERVER ?= docker.io
DOCKER_REGISTRY ?= $(DOCKER_SERVER)/kumahq
```

Reply:
What would break? It would just be the hostname part of the URL, right? So `foo.com` here.

I introduced https://github.com/kumahq/kuma/pull/6602 precisely for the case that the registry is just a hostname, there doesn't have to be a path component in the registry. So when an image looks like `something.com/kuma-cp`.

I'm going to change to the make style substitution.

Reply:
got it

Reply:
Ah I see what you mean. Then we can still just set DOCKER_REGISTRY